### ISSUE TITLE
"custom-checkbox" vertical alignment not working when base font size changed

### ISSUE THREAD

**Issue by kikonen** [twbs/bootstrap#25828]
bootstrap 4.0.0:

Changed base font size to be smaller

```
$font-size-base:              0.75rem !default;
```

After this vertical alignment in custom-control is clearly mis-aligned between checkbox and label (label should be few pixels more down vertically).

```
<div class="custom-control custom-checkbox">
  <input class="custom-control-input" type="checkbox" value="true">
  <label class="custom-control-label">Anonymous</label>
</div>
```

**Comment by Johann-S** [comment]
Bug reports must include a live demo of the problem. Per our [contributing guidelines](https://github.com/twbs/bootstrap/blob/v4-dev/.github/CONTRIBUTING.md), please create a reduced test case via [CodePen](https://codepen.io/) or [JS Bin](https://jsbin.com/) and report back with your link, Bootstrap version, and specific browser and OS details.

**Comment by kikonen** [comment]
Hmm... not seeing direct way how to get bootstrap SCSS into Codepen so that could do "changing base font size to be smaller" in SCSS variables.

As workaround did:
````
  .custom-control-label {
    margin-top: 4px;
  }
```

**Comment by istratos** [comment]
https://codepen.io/anon/pen/YaWGKZ

What I found is that the circle of the radius control, and the check in the checkbox, are misaligned when the font is smaller (13px and 14px) or bigger (17px and 18px), haven't tried other sizes but my guess it would also happen with weird sizes.

**Comment by bakedbird** [comment]
The issue is caused because rem unit, which is used by bootstrap, is relative to the font-size of the root (html) element. This means that if $font-size-base is applied on the html tag (maybe on reset.scss) the issue will be solved.

Note that the issue that @istratos mentioned is valid *if* the font-size applied on the html tag is based using px (which is what many people still use).

**Comment by asyncdesign** [comment]
The problem is not caused by using rem units - the issue is how the browser calculates the background position of the after pseudo element when using a combination of absolute top 0.25rem, background-size: 50% 50%, and background-position: center center. This is how the SVG image is positioned. Also, it doesn't help when the before pseudo element has an absolute top of 0.25rem. Browsers just can't render all of this accurately at various font sizes, but at least it scales. If there is a better way it would be interesting to know about.

**Comment by mdo** [comment]
Fixed by #25602.
